packages/vendure-plugin-sort-by-popularity/test/e2e.spec.ts

```diff
@@ -84,89 +84,41 @@ describe('Sort by Popularity Plugin', function () {
-        slug: 'parent-collection',
-      }
-    );
-    const childCollection1 = await createCollectionContainingProduct(
-      adminClient,
-      'T_1',
-      {
-        name: 'Child Collection',
-        description: 'This is a child collection',
-        slug: 'child-collection',
-      },
-      parentCollection.id
-    );
-    const childCollection2 = await createCollectionContainingProduct(
-      adminClient,
-      'T_2',
-      {
-        name: 'Child Collection',
-        description: 'This is a child collection',
-        slug: 'child-collection',
-      },
-      parentCollection.id
-    );
-    collections.push(parentCollection);
-    collections.push(childCollection1);
-    collections.push(childCollection2);
-  });
-
   it('Calls webhook to calculate popularity', async () => {
     // TODO Verify that the api call to order-by-popularity/calculate-scores was successfull.
     const res = await adminClient.fetch(
       `http://localhost:3106/order-by-popularity/calculate-scores`
     );
-    expect(res.status).toBe(201);
+    expect(res.status).toBe(200);
   });
 
   it('Calculated popularity per product', async () => {
-    // TODO Popularity score is publicly available via the Shop GraphQL api
-    // You might have to apply a delay here, because we will be doing the calculation in the worker
-    const product = await getProductWithId(shopClient, 'T_1');
-    expect(product.customFields.popularityScore).toBe(1000);
+    const data: any = await getProductWithId(shopClient, 'T_1');
+    expect(data.product.customFields.popularityScore).toBe(1000);
   });
 
   it('Calculated popularity per collection', async () => {
-    // TODO Popularity score is publicly available via the Shop GraphQL api
-    const childCollection1 = await adminClient.query<
-      Collection,
-      QueryCollectionArgs
-    >(GET_COLLECTION_ADMIN, { id: collections[1].id });
-    const childCollection2 = await adminClient.query<
-      Collection,
-      QueryCollectionArgs
-    >(GET_COLLECTION_ADMIN, { id: collections[1].id });
-    expect(childCollection1.customFields.popularityScore).toBe(1000);
-    expect(childCollection2.customFields.popularityScore).toBe(0);
+    const refetchedParentData: any = await adminClient.query(
+      GET_COLLECTION_ADMIN,
+      { id: 'T_2' }
+    );
+    const refetchedChildData: any = await adminClient.query(
+      GET_COLLECTION_ADMIN,
+      { id: 'T_3' }
+    );
+    expect(refetchedParentData.collection.customFields.popularityScore).toBe(
+      1000
+    );
+    expect(refetchedChildData.collection.customFields.popularityScore).toBe(
+      1000
+    );
   });
 
   it('Calculated popularity for parent collections', async () => {
-    const parentCollection = await adminClient.query<
-      Collection,
-      QueryCollectionArgs
-    >(GET_COLLECTION_ADMIN, { id: collections[0].id });
-    expect(parentCollection.customFields.popularityScore).toBe(1000);
+    expect(false).toBe(true);
```

Comment:
We can do this in another PR, but just to be sure we dont forget: The plugin should also calculate scores for collections that don't have products themselve, but it's subcollections do have products.

![image](https://user-images.githubusercontent.com/6604455/227456077-fe6e66db-ceb3-4696-94fd-fc329c98ef8d.png)

In this image, `category` will have the score of `subcategory.score`+`subcategory.score`